docs/deprecation-process.md

```diff
@@ -0,0 +1,422 @@
+# Deprecation process
+
+## Motivation
+
+Deprecation of code is a difficult topic,
+which can become a point of friction.
+
+On one hand, maintainers have to maintain the code base
+for many platforms, many compilers, while making bug fixes
+and adding new features over time.
+
+To do so, sometime code cleanup, minor refactoring, or removal of
+previous engineering debt is required to maintain a healthy code base,
+to allow for further improvements.
+
+In short, maintainers may want to, or may have to,
+remove some apis during the lifetime of a project.
+
+On the other hand, users who have a working application built using
+opentelemetry-cpp, for a given platform, a given compiler, and a given
+configuration, prefer to have stability and as few changes as possible.
+
+Doing code changes in the application imply a new build / test / deploy
+cycle, which induce costs, and risks.
+
+In short, users typically do now want changes, if the code works for them.
+
+The following extreme behaviors are what needs to be avoided for a project
+to stay healthy:
+
+* constant changes in the code base, done without notice,
+  introducing incompatibilities, causing adoption of each new release to
```

Comment:
Also mention that different code are handled differently, like incompatibility of preview API/SDK is acceptable like for logs for not for stable API/SDK？

Reply:
Thanks, good idea.

Fixed.

Reply:
Actually, we can mention that in case deprecation is having ABI breaking changes for API for stable feature, the  `OPENTELEMETRY_ABI_VERSION_NO` would be incremented to next value.